### ISSUE TITLE
completion of the server MVP

### ISSUE THREAD

**Issue by RebelOfDeath** [AISE-TUDelft/coco#10]
finalization of #8 and #9 with additional testing to minimize the likelihood of bugs.

**Comment by Ar4l** [comment]
Ideally we also include #11

**Comment by RebelOfDeath** [comment]
@Ar4l, agreed!
I'll pick up #11 as I'll be working on the endpoints for the time being so I can also look into #12 when doing so. I think something we could potentially look into when we are also implementing sessions is to have a session token that is exchanged and that instead of users passing their actual user token, they communicate with the server using their session token.

This would also allow us to do some cool stuff down the line with tracking user behavior patterns within each session.

**Comment by RebelOfDeath** [comment]
I'll add #11 and #12 to the sprint targets for server MVP

**Comment by Ar4l** [comment]
Yes session tokens are a must because it allows us to rate-limit the authentication API (in the event of ddos) while ensuring users that are in-session can continue.

**Comment by RebelOfDeath** [comment]
Essentially besides some modification that still needs to happen on the LLM side of things with regards to #7  (which for now is mocked in the tests) I see no blocking factor in starting to write the actual plugins as of 646a3c57084eede0472e90fbe20d6bfa571ab39a.

**Comment by Ar4l** [comment]
Added the need to dockerise, so we can just pull and deploy.
